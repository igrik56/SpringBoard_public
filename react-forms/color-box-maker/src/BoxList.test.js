import React from "react";
import {render, fireEvent} from '@testing-library/react'
import BoxList from "./BoxList";

function addBox (boxList, height=100, width = 100, color = 'black'){

    const heightInput = boxList.getByLabelText('Height')
    const widthInput = boxList.getByLabelText('Width')
    const backgroundColorInput = boxList.getByLabelText('Background Color')
    const button = boxList.getByText('Add a box')

    fireEvent.change(heightInput, {target: {value: height}})
    fireEvent.change(widthInput, {target: {value: width}})
    fireEvent.change(backgroundColorInput, {target: {value: color}})
    fireEvent.click(button)
}

test('smoke test BoxList', () => {
    render(<BoxList />);
  });
  
it('snapshot test BoxList', function(){
const {asFragment} = render(<BoxList/>)
expect (asFragment()).toMatchSnapshot()
})


it('adding a new box', function (){
    const boxList = render(<BoxList/>)

    //checking if there is any box. At start has to be empty.
    expect(boxList.queryByText('X')).not.toBeInTheDocument()

    //adding a box
    addBox(boxList)

    //expecting to see the default box. 
    //Looking for the x button. 
    //Cause it is the only text that is in the box.

    const removeButton = boxList.getByText('X')

    expect(removeButton).toBeInTheDocument()

    expect(removeButton.parentElement.parentElement).toHaveStyle(`
        height: 100px;
        width: 100px;
        background-color: black;`)
})

it('remove a box', function (){
    const boxList = render(<BoxList />)
    addBox(boxList)

    //checking if the box was created.
    const removeButton = boxList.getByText('X')
    expect(removeButton).toBeInTheDocument()    
    
    //clicking the button and expecting the box to be removed.
    fireEvent.click(removeButton)
    expect(removeButton).not.toBeInTheDocument()
})