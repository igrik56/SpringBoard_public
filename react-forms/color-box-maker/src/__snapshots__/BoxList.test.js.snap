// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test BoxList 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <div>
          <label
            for="height"
          >
            Height
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="width"
          >
            Width
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="backgroundColor"
          >
            Background Color
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            type="text"
            value=""
          />
        </div>
        <button
          id="newBoxBtn"
        >
          Add a box
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
